src/extract/IExtractor.h

```diff
@@ -37,6 +41,7 @@ class IExtractor
   }
 
 public:
+  IExtractor(const vuplus::Settings &settings) : m_settings(settings) {};
```

Comment:
Please put the public section at the beginning of the class decl.
Please consolidate the two protected sections to one, located after the public section.